sway/desktop/render.c

```diff
@@ -45,9 +46,29 @@ struct decoration_data get_undecorated_decoration_data() {
 		.corner_radius = 0,
 		.saturation = 1.0f,
 		.has_titlebar = false,
+		.blur = false,
 	};
 }
 
+int get_blur_size() {
+	return pow(2, config->blur_params.num_passes) * config->blur_params.radius;
+}
+
+int get_container_expanded_size(struct sway_container *con) {
+	bool shadow_enabled = con ? con->shadow_enabled : config->shadow_enabled;
+	int shadow_sigma = shadow_enabled ? config->shadow_blur_sigma : 0;
+
+	bool blur_enabled = con ? con->blur_enabled : config->blur_enabled;
+	int blur_size = blur_enabled ? get_blur_size() : 0;
+
+	// +1 as a margin of error
+	return MAX(shadow_sigma, blur_size) + 1;
+}
```

Comment:
This is currently only ever called once in render.c, and a value of NULL is passed to it. We should refactor or remove this function